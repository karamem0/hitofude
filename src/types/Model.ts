//
// Copyright (c) 2023 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

export interface File {
  id: string,
  baseName?: string,
  fullName?: string,
  createdDate?: Date,
  updatedDate?: Date,
  downloadUrl?: string,
  parentId?: string
}

export interface FileContent {
  content: string,
  editable?: boolean
}

export interface Folder {
  id: string,
  name?: string,
  createdDate?: Date,
  updatedDate?: Date,
  root?: boolean,
  webUrl?: string,
  parentId?: string,
  folders?: Folder[],
  files?: File[]
}

export interface DialogAction {
  type: DialogType,
  payload: unknown
}

export enum DialogType {
  copyFile = 'copyFile',
  createFile = 'createFile',
  createFolder = 'createFolder',
  deleteFile = 'deleteFile',
  deleteFolder = 'deleteFolder',
  renameFile = 'renameFile',
  renameFolder = 'renameFolder'
}

export interface TabMode {
  type?: TabType,
  open?: boolean
}

export enum TabType {
  explorer = 0,
  search = 1,
}
