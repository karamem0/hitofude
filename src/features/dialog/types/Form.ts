//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

export interface FileCopyDialogFormState {
  baseName?: string
}

export interface FileCreateDialogFormState {
  baseName?: string
}

export interface FileRenameDialogFormState {
  id: string,
  baseName?: string
}

export interface FolderCreateDialogFormState {
  name?: string
}

export interface FolderRenameDialogFormState {
  id: string,
  name?: string
}
