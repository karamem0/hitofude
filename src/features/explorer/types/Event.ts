//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import { FileRejection } from 'react-dropzone';

export interface DropEventData {
  acceptedFiles: File[],
  rejectedFiles: FileRejection[]
}
