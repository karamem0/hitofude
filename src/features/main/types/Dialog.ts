//
// Copyright (c) 2023 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

export enum DialogType {
  copyFile = 'copyFile',
  createFile = 'createFile',
  createFolder = 'createFolder',
  deleteFile = 'deleteFile',
  deleteFolder = 'deleteFolder',
  renameFile = 'renameFile',
  renameFolder = 'renameFolder'
}

export interface DialogAction {
  type: DialogType,
  payload: unknown
}
