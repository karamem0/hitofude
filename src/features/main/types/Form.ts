//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

export interface SearchTabItemFormState {
  query?: string
}
