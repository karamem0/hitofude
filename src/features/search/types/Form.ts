//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

export interface SearchTabPanelFormState {
  query?: string
}

export type SearchTabPanelFormField = keyof SearchTabPanelFormState;
