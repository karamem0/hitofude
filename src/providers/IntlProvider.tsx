//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import { IntlProvider as Provider, createIntl } from 'react-intl';
import en from '../translations/en.json';
import ja from '../translations/ja.json';

const translations: { [key: string]: Record<string, string> } = {
  en,
  ja
};

const intl = createIntl({
  defaultLocale: 'en',
  locale: window.navigator.language.substring(0, 2),
  messages: translations[window.navigator.language]
});

function IntlProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  return (
    <Provider {...intl}>
      {children}
    </Provider>
  );

}

export default IntlProvider;
