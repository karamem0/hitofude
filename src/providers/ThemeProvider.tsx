//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import {
  FluentProvider as Provider,
  Theme
} from '@fluentui/react-components';

import { darkTheme, lightTheme } from '../themes/Theme';
import { InvalidOperationError } from '../types/Error';
import { ThemeName } from '../types/Model';

interface ThemeState {
  theme: Theme,
  themeName: ThemeName
}

interface ThemeContextState {
  theme: ThemeState,
  changeTheme: (themeName: ThemeName) => void
}

const ThemeContext = React.createContext<ThemeContextState | undefined>(undefined);

export const useTheme = (): ThemeContextState => {
  const value = React.useContext(ThemeContext);
  if (value == null) {
    throw new InvalidOperationError();
  }
  return value;
};

function ThemeProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  const [ theme, setTheme ] = React.useState<ThemeState>({
    theme: lightTheme,
    themeName: ThemeName.light
  });

  const handleChangeTheme = React.useMemo(() => (value: ThemeName) => {
    switch (value) {
      case ThemeName.light:
        setTheme(() => ({
          theme: lightTheme,
          themeName: value
        }));
        break;
      case ThemeName.dark:
        setTheme(() => ({
          theme: darkTheme,
          themeName: value
        }));
        break;
      default:
        break;
    }
  }, []);

  const value = React.useMemo<ThemeContextState>(() => ({
    theme,
    changeTheme: handleChangeTheme
  }), [
    theme,
    handleChangeTheme
  ]);

  return (
    <ThemeContext.Provider value={value}>
      <Provider theme={value.theme.theme}>
        {children}
      </Provider>
    </ThemeContext.Provider>
  );

}

export default ThemeProvider;
