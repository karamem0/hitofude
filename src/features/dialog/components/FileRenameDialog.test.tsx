//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import { render, screen } from '@testing-library/react';
import { File } from '../../../types/Model';
import IntlProvider from '../../../providers/IntlProvider';
import Presenter from './FileRenameDialog.presenter';
import ThemeProvider from '../../../providers/ThemeProvider';
import userEvent from '@testing-library/user-event';

vi.mock('../../../common/components/ModalDialog', () => ({
  default: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="test-ModalDialog">
      {children}
    </div>
  )
}));

it('should match the snapshot when the loading is true', () => {
  const container = document.body.appendChild(document.createElement('div'));
  const params = {
    loading: true,
    mountNode: container,
    value: {
      id: '01f02f1e-fcfa-4629-aab1-c9fad7efe536',
      baseName: 'README',
      extension: 'md'
    } as File
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>,
    {
      container
    }
  );
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTitle('Save')).toBeDisabled();
});

it('should match the snapshot when the loading is false', async () => {
  const user = userEvent.setup();
  const container = document.body.appendChild(document.createElement('div'));
  const params = {
    loading: false,
    mountNode: container,
    value: {
      id: '01f02f1e-fcfa-4629-aab1-c9fad7efe536',
      baseName: 'README',
      extension: 'md'
    } as File
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>,
    {
      container
    }
  );
  await user.click(screen.getByPlaceholderText('File name'));
  await user.keyboard('1');
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByTitle('Save')).not.toBeDisabled();
});
