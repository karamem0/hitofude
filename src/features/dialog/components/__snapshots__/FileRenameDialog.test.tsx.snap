// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot when the loading is false 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fui-DialogSurface__backdrop r1e18s3l"
  />
  .emotion-0 {
  overflow: hidden;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 1rem 0;
}

<div
    aria-modal="true"
    class="fui-DialogSurface rsmdyd3"
    role="dialog"
    tabindex="-1"
  >
    <form>
      <div
        class="fui-DialogBody r1h3qql9"
      >
        <h2
          class="fui-DialogTitle rxjm636 ___mh25s00_kwrsjh0 fsyjsko"
        >
          Rename file
        </h2>
        <div
          class="fui-DialogContent r1v5zwsm emotion-0"
        >
          <div
            class="emotion-1"
          >
            <span
              class="fui-Input r1oeeo9n ___xay2ks0_8jjmwj0 f11gcy0p fvcxoqz f1ub3y4t f1l4zc64 f1m52nbi f8vnjqi fz1etlk f1klwx88 f1hc16gm"
            >
              <input
                aria-label="File name"
                class="fui-Input__input r12stul0 ___1q6bera_bzdj540 ffczdla"
                placeholder="File name"
                type="text"
                value="README1"
              />
              <span
                class="fui-Input__contentAfter r1572tok"
              >
                .md
              </span>
            </span>
          </div>
        </div>
        <div
          class="fui-DialogActions rhfpeu0 ___pj8zjl0_1y84z0u f1a7i8kp fd46tj4 fsyjsko f1f41i0t f1jaqex3 f2ao6jk"
        >
          <button
            aria-label="Save"
            class="fui-Button r1alrhcs ___1akj6hk_ih97uj0 ffp7eso f1p3nwhy f11589ue f1q5o8ev f1pdflbu f1phragk f15wkkf3 f1s2uweq fr80ssc f1ukrpxl fecsdlb f1rq72xc fnp9lpt f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1d6v5y2 f1rirnrt f1uu00uk fkvaka8 f1ux7til f9a0qzu f1lkg8j3 fkc42ay fq7113v ff1wgvm fiob0tu f1j6scgf f1x4h75k f4xjyn1 fbgcvur f1ks1yx8 f1o6qegi fcnxywj fmxjhhp f9ddjv3 f17t0x8g f194v5ow f1qgg65p fk7jm04 fhgccpy f32wu9k fu5nqqq f13prjl2 f1czftr5 f1nl83rv f12k37oa fr96u23"
            title="Save"
            type="submit"
          >
            Save
          </button>
          <button
            aria-label="Cancel"
            class="fui-Button r1alrhcs"
            data-tabster="{"restorer":{"type":1}}"
            title="Cancel"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </form>
  </div>
  <div
    class="fui-FluentProvider fui-FluentProviderr2 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      data-testid="test-ModalDialog"
    >
      <span
        hidden=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`should match the snapshot when the loading is true 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fui-DialogSurface__backdrop r1e18s3l"
  />
  .emotion-0 {
  overflow: hidden;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 1rem 0;
}

<div
    aria-modal="true"
    class="fui-DialogSurface rsmdyd3"
    role="dialog"
    tabindex="-1"
  >
    <form>
      <div
        class="fui-DialogBody r1h3qql9"
      >
        <h2
          class="fui-DialogTitle rxjm636 ___mh25s00_kwrsjh0 fsyjsko"
        >
          Rename file
        </h2>
        <div
          class="fui-DialogContent r1v5zwsm emotion-0"
        >
          <div
            class="emotion-1"
          >
            <span
              class="fui-Input r1oeeo9n ___xay2ks0_8jjmwj0 f11gcy0p fvcxoqz f1ub3y4t f1l4zc64 f1m52nbi f8vnjqi fz1etlk f1klwx88 f1hc16gm"
            >
              <input
                aria-label="File name"
                class="fui-Input__input r12stul0 ___1q6bera_bzdj540 ffczdla"
                placeholder="File name"
                type="text"
                value="README"
              />
              <span
                class="fui-Input__contentAfter r1572tok"
              >
                .md
              </span>
            </span>
          </div>
        </div>
        <div
          class="fui-DialogActions rhfpeu0 ___pj8zjl0_1y84z0u f1a7i8kp fd46tj4 fsyjsko f1f41i0t f1jaqex3 f2ao6jk"
        >
          <button
            aria-label="Save"
            class="fui-Button r1alrhcs ___lwxuac0_1ursfbr f1bg9a2p f1p3nwhy f11589ue f1q5o8ev f1pdflbu f1s2aq7o f1falr9n f1s2uweq fr80ssc f1ukrpxl fecsdlb fvgxktp f1t6o4dc f1h0usnq fs4ktlq f16h9ulv fx2bmrt fcvwxyo f4lkoma fg455y9 f1rvyvqg f1cwzwz f14g86mu f1dcs8yz fkc42ay f1vmkb5g f53ppgq f1663y11 f18v5270 f80fkiy f1kc2mi9 f1y0svfh fihuait fnxhupq fx507ft fyd6l6x fb3rf2x fdrzuqr f15x8b5r fphbwmw f19vpps7 fv5swzo f1al02dq f8w4c43 f1ol4fw6 f1q1lw4e f1dwjv2g fuigjrg fjwq6ea f1lr3nhc fn5gmvv f1mbxvi6 f4dhi0o fequ9m0 f17t0x8g f194v5ow f1qgg65p fk7jm04 fhgccpy f32wu9k fu5nqqq f13prjl2 f1czftr5 f1nl83rv f12k37oa fr96u23"
            disabled=""
            title="Save"
            type="submit"
          >
            Save
          </button>
          <button
            aria-label="Cancel"
            class="fui-Button r1alrhcs"
            data-tabster="{"restorer":{"type":1}}"
            title="Cancel"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </form>
  </div>
  <div
    class="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      data-testid="test-ModalDialog"
    >
      <span
        hidden=""
      />
    </div>
  </div>
</DocumentFragment>
`;
